homeassistant/components/device_tracker/life360.py

```diff
@@ -0,0 +1,145 @@
+"""
+Support for the Life360 platform.
+
+For more details about this platform, please refer to the documentation at
+TODO
+"""
+import logging
+
+#making http requests happening
+import requests
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+from homeassistant.const import CONF_USERNAME, CONF_PASSWORD
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import track_utc_time_change
+from homeassistant.util import slugify
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+
+def setup_scanner(hass, config: dict, see, discovery_info=None):
+    """Validate the configuration and return a Life360 scanner."""
+    Life360Scanner(hass, config, see)
+    return True
+
+
+class Life360Scanner(object):
+    """A class representing a Life360 scanner."""
+
+    def __init__(self, hass, config: dict, see) -> None:
+        """Initialize the Life360 scanner."""
+
+        self.hass = hass
+        self.see = see
+
+        self.username = config.get(CONF_USERNAME)
+        self.password = config.get(CONF_PASSWORD)
+
+        self._update_info()
+
+        track_utc_time_change(
+            self.hass, self._update_info, second=range(0, 60, 30))
+    
+    def get_members(self):
+        #reset member to nothing
+        self.members = []
+
+        #get bearer token
+        url = 'https://api.life360.com/v3/oauth2/token.json'
+        payload = payload = {'grant_type': 'password', 'username': self.username, 'password': self.password}
+        headers = {'Authorization': 'Basic cFJFcXVnYWJSZXRyZTRFc3RldGhlcnVmcmVQdW1hbUV4dWNyRUh1YzptM2ZydXBSZXRSZXN3ZXJFQ2hBUHJFOTZxYWtFZHI0Vg=='}
+        r = requests.post(url, data=payload , headers=headers)
+
+        #check if we have valid response
+        if r.status_code != 200:
+            _LOGGER.error("Didn't get a good http response while logging in: %s", r.status_code)
+            return None
+
+        #hopefully this is working
+        try:
+            data = r.json()
+            self.access_token = data['access_token']
+        except:
+            _LOGGER.error("Failed to Login")
+            self.success_init = False
+
+        _LOGGER.info("Scanning")
+
+        """
+        Life360 has the following hierarchy to list devices
+        Account -> Circles -> Members enrolled in Circle
+        """
+        #first retrieve the circles
+        url = "https://api.life360.com/v3/circles.json"
+        headers = {'Content-Type': 'application/json','Authorization': 'Bearer ' + self.access_token}
```

Comment:
Replace key and values with constants where possible.

Reply:
Hi fabaff I didn't quite get what you meant with this comment. How can I fix this?